[2.38] Taking some IOS GC branches for WPE as well

tomasz-karczewski-red opened the pull request:
The aim is to make the GC more reliable on low-memory platforms. IOS branches applied:

- FullGCActivityCallback::doCollection - do not bail out of GC early
- MemoryPressureHandler - apply IOS defaults for conservative/strict mode thresholds
- MemoryRelease / releaseCriticalMemory - request immediate GC
- WorkerGlobalScope::deleteJSCodeAndGC - collect even if called with Synchronous::No

magomez commented on Source/WTF/wtf/MemoryPressureHandler.cpp:
```diff
@@ -40,7 +40,7 @@ namespace WTF {
 WTF_EXPORT_PRIVATE bool MemoryPressureHandler::ReliefLogger::s_loggingEnabled = false;
 
-#if PLATFORM(IOS_FAMILY)
+#if PLATFORM(IOS_FAMILY) || PLATFORM(WPE)
```
We cannot change the default values of the fractions, cause this affects all of the devices. As I told you in previous PRs, there's API to change these values for your platform only, please use it.

magomez reviewed (changes requested):
Please, we have here 4 changes that should go in 4 separate commits. Well, 3, as the change in the MemoryPressureHandler fractions should not be there. Please, remove the fractions change and split the PR in 3 different commits.

tomasz-karczewski-red committed:
FullGCActivityCallback: take IOS GC branch for WPE

The aim is to make the GC more reliable on low-memory platforms.
In FullGCActivityCallback, the change is
to do not bail out of GC early.

tomasz-karczewski-red committed:
MemoryRelease: take IOS GC branch for WPE

The aim is to make the GC more reliable on low-memory platforms.
In MemoryRelease, the change is to request immediate GC in
releaseCriticalMemory.

tomasz-karczewski-red committed:
WorkerGlobalScope: take IOS GC branch for WPE

The aim is to make the GC more reliable on low-memory platforms.
In WorkerGlobalScope::deleteJSCodeAndGC the change is to
collect even if called with 'Synchronous::No'

tomasz-karczewski-red commented:
sure - I've just updated the PR

woutermeek commented:
Thanks @magomez, we'll investigate if there are any performance issues.
